src/util/time.cpp

```diff
@@ -6,3 +6,33 @@ LLTIMER Time::s_timer;
 bool Time::s_testMode = false;
 // static
 qint64 Time::s_testElapsed_nsecs = 0;
+
+// static
+QString Time::formatSeconds(double dSeconds, bool showCentis) {
+    if (dSeconds < 0) {
+        return "?";
+    }
+
+    const int days = static_cast<int>(dSeconds) / kSecondsPerDay;
+    dSeconds -= days * kSecondsPerDay;
+
+    // NOTE(uklotzde): Time() constructs a 'null' object, but
+    // we need 'zero' here.
+    QTime t = QTime(0, 0).addMSecs(dSeconds * kMillisPerSecond);
+
+    QString formatString =
+            (days > 0 ? (QString::number(days) %
+                         QLatin1String("'d', ")) : QString()) %
+            QLatin1String(days > 0 || t.hour() > 0 ? "hh:mm:ss" : "mm:ss") %
+            QLatin1String(showCentis ? ".zzz" : "");
```

Comment:
QLocale::decimalPoint() is a separate issue.

But I added a unit test and it turns out that the formatting algorithm is 
wrong! Fix and tests will follow.

On 01/08/2016 09:47 PM, Daniel Schürmann wrote:
>
> In src/util/time.cpp 
> <https://github.com/mixxxdj/mixxx/pull/845#discussion_r49236294>:
>
> > +    if (dSeconds < 0) {
> > +        return "?";
> > +    }
> > +
> > +    const int days = static_cast<int>(dSeconds) / kSecondsPerDay;
> > +    dSeconds -= days * kSecondsPerDay;
> > +
> > +    // NOTE(uklotzde): Time() constructs a 'null' object, but
> > +    // we need 'zero' here.
> > +    QTime t = QTime(0, 0).addMSecs(dSeconds * kMillisPerSecond);
> > +
> > +    QString formatString =
> > +            (days > 0 ? (QString::number(days) %
> > +                         QLatin1String("'d', ")) : QString()) %
> > +            QLatin1String(days > 0 || t.hour() > 0 ? "hh:mm:ss" : "mm:ss") %
> > +            QLatin1String(showCentis ? ".zzz" : "");
>
> Unrelated, but I have just noticed it:
> http://stackoverflow.com/questions/31477309/timestamp-in-iso-8601-the-last-6-digits-yyyy-mm-ddthhmmss
> it should be QLocale::decimalPoint()
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/845/files#r49236294>.
>